roofit/roofitcore/src/RooConvGenContext.cxx

```diff
@@ -142,17 +138,24 @@ RooConvGenContext::RooConvGenContext(const RooNumConvPdf &model, const RooArgSet
          << " for generation of observable(s) " << vars << endl ;
 
   // Create generator for physics X truth model
-  _pdfVarsOwned = (RooArgSet*) model.conv().clonePdf().getObservables(&vars)->snapshot(true) ;
-  _pdfVars = new RooArgSet(*_pdfVarsOwned) ;
-  _pdfGen = ((RooAbsPdf&)model.conv().clonePdf()).genContext(*_pdfVars,prototype,auxProto,verbose) ;
-  _pdfCloneSet = 0 ;
+  {
+    RooArgSet clonedPdfObservables;
+    model.conv().clonePdf().getObservables(&vars, clonedPdfObservables);
+    _pdfVarsOwned.reset(clonedPdfObservables.snapshot(true));
+  }
+  _pdfVars = std::make_unique<RooArgSet>(*_pdfVarsOwned) ;
+  _pdfGen.reset(static_cast<RooAbsPdf&>(model.conv().clonePdf()).genContext(*_pdfVars,prototype,auxProto,verbose));
 
   // Create generator for resolution model as PDF
-  _modelVarsOwned = (RooArgSet*) model.conv().cloneModel().getObservables(&vars)->snapshot(true) ;
-  _modelVars = new RooArgSet(*_modelVarsOwned) ;
+  {
+    RooArgSet clonedModelObservables;
+    model.conv().cloneModel().getObservables(&vars, clonedModelObservables);
```

Comment:
This is the over line where there was a leak from the call to `getObservables()` before.